src/Workspaces/Core/Portable/Utilities/Operations/IOperationContextFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// Factory to create <see cref="IOperationContext"/>s.
+    /// </summary>
+    internal interface IOperationContextFactory : IWorkspaceService
```

Comment:
intentionally not public.  Users will be handed context/scopes.  They will not create them themselves.